### ISSUE TITLE
Allow commenting of attributes

### ISSUE THREAD

**Issue by sampattuzzi** [pugjs/jade#2209]
Obviously commenting is supported with either `//` or `//-` and this allows only line or block comments. However, it's impossible to comment out attributes using this scheme even if they are split onto a separate line. I'm guessing this is a bug with the grammar.

I have tested this with 1.11.0 so there may be a fix since then.

# Example.
This code doesn't comment out the type attribute but ignores the `//-` symbols
```
input(
  //-type='checkbox'
  name='agreement'
  checked
)
```

**Comment by ForbesLindesay** [comment]
I completely agree that we should make this work.  We currently don't look for jade/pug style comments in attributes.  You can use javascript style block comments after the first attribute.  e.g.

```
input(
  type='checkbox'
  /*name='agreement'*/
  checked
)
```

but that's useless for the first argument, and a lot more faffy than just a line comment.

**Comment by TimothyGu** [comment]
I had no idea that this worked…

But you could also do this:

```jade
input(
  type=false /* 'checkbox' */
  name='agreement'
  checked
)
```

No argument here on it being as ugly as heck.

**Comment by tomByrer** [comment]
I'm wondering if there should be a pre-filter of comments before the Jade compiler?

**Comment by TimothyGu** [comment]
@tomByrer a step similar to that has been added in #2043. However, unbuffered comment is a very special case, since we still treats it as a first-class token, and does not mix well with other tokens.

**Comment by Goddak** [comment]
Did this ever get fixed?
